backend/src/packages/comments/comment.service.ts

```diff
@@ -0,0 +1,86 @@
+import { ApplicationError } from '~/libs/exceptions/exceptions.js';
+import { type IService } from '~/libs/interfaces/service.interface.js';
+
+import { CommentEntity } from './comment.entity.js';
+import { type CommentRepository } from './comment.repository.js';
+import {
+  type CommentBaseResponseDto,
+  type CommentCreateDto,
+  type CommentGetAllResponseDto,
+  type CommentUpdateRequestDto,
+} from './libs/types/types.js';
+
+class CommentService implements IService {
+  private commentRepository: CommentRepository;
+
+  public constructor(commentRepository: CommentRepository) {
+    this.commentRepository = commentRepository;
+  }
+
+  public findAll(): Promise<{ items: unknown[] }> {
+    return Promise.resolve({ items: [] });
+  }
+
+  public async findAllByArticleId(
+    articleId: number,
+  ): Promise<CommentGetAllResponseDto | []> {
+    const items = await this.commentRepository.findAllByArticleId(articleId);
+
+    return {
+      items: items.map((it) => it.toObject()),
+    };
+  }
+
+  public async find(id: number): Promise<CommentBaseResponseDto | null> {
+    const comment = await this.commentRepository.find(id);
+
+    if (!comment) {
+      return null;
+    }
+
+    return comment.toObject();
+  }
+
+  public async create(
+    payload: CommentCreateDto,
+  ): Promise<CommentBaseResponseDto> {
+    const comment = await this.commentRepository.create(
+      CommentEntity.initializeNew({
+        text: payload.text,
+        userId: payload.userId,
+        articleId: payload.articleId,
+        publishedAt: payload?.publishedAt ?? null,
+      }),
+    );
+
+    return comment.toObject();
+  }
+
+  public async update(
+    id: number,
+    payload: CommentUpdateRequestDto,
+  ): Promise<CommentBaseResponseDto> {
+    const comment = await this.find(id);
+
+    if (!comment) {
+      throw new ApplicationError({
+        message: `Comment with id ${id} not found`,
+      });
+    }
+
+    const updatedComment = await this.commentRepository.update(
+      CommentEntity.initialize({
+        ...comment,
+        ...payload,
+      }),
+    );
+
+    return updatedComment.toObject();
+  }
+
+  public async delete(id: number): Promise<boolean> {
+    return await this.commentRepository.delete(id);
```

Comment:
I guess there's no need to await promise here, we can simply return it without awaiting